Geometry/MTDGeometryBuilder/interface/MTDGeometry.h

```diff
@@ -0,0 +1,93 @@
+#ifndef Geometry_MTDGeometryBuilder_MTDGeometry_H
+#define Geometry_MTDGeometryBuilder_MTDGeometry_H
+
+#include "Geometry/CommonDetUnit/interface/TrackingGeometry.h"
+#include "Geometry/CommonDetUnit/interface/GeomDetEnumerators.h"
+#include "Geometry/CommonDetUnit/interface/MTDGeomDet.h"
+
+class GeometricTimingDet;
+
+/**
+ * A specific MTD Builder which builds MTD from a list of DetUnits. 
+ * Pattern recognition is used to discover layers, rings etc.
+ */
+class MTDGeometry final : public TrackingGeometry {
+
+  explicit MTDGeometry(GeometricTimingDet const* gd=nullptr);  
+  
+  friend class MTDGeomBuilderFromGeometricTimingDet;
+
+  void addType(GeomDetType const * p);
+  void addDetUnit(GeomDet const * p);
+  void addDetUnitId(DetId p);
+  void addDet(GeomDet const * p);
+  void addDetId(DetId p);
+  void finalize();
+
+public:
+  typedef GeomDetEnumerators::SubDetector SubDetector;
+
+  enum class ModuleType {
+    UNKNOWN, 
+      BTL, 
+      ETL, 
+   };
+
+  ~MTDGeometry() override ;
+
+  const DetTypeContainer&  detTypes()         const override {return theDetTypes;}
+  const DetContainer&      detUnits()         const override {return theDetUnits;}
+  const DetContainer&      dets()             const override {return theDets;}
```

Comment:
DetUnits are the individual detectors, dets() is the list of all detectors + composite layers. Allows a simultaneous record keeping so that you can build faster geometrys for navigation and provide all necessary coordinate transforms.